src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+                authority_name: ME_AUTHORITY.to_string(), // TODO: Should we set this to anything specific?
+                ue_id: service_id as u32,
+                ue_version_major: interface_version as u32,
+                resource_id: method_id as u32,
+                ..Default::default()
+            };
+
+            let umsg_res = UMessageBuilder::publish(source).build();
+
+            let Ok(umsg) = umsg_res else {
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    format!(
+                        "Unable to build UMessage from vsomeip message: {:?}",
+                        umsg_res.err().unwrap()
+                    ),
+                ));
+            };
+
+            Ok(umsg)
+        }
+        message_type_e::MT_RESPONSE => {
+            trace!("MT_RESPONSE type");
+            let sink = UUri {
+                authority_name,
+                ue_id: client_id as u32,
+                ue_version_major: 1, // TODO: I don't see a way to get this
+                resource_id: 0,      // set to 0 as this is the resource_id of "me"
+                ..Default::default()
+            };
+
+            let source = UUri {
+                authority_name: ME_AUTHORITY.to_string(), // TODO: Should we set this to anything specific?
+                ue_id: service_id as u32,
+                ue_version_major: interface_version as u32,
+                resource_id: method_id as u32,
+                ..Default::default()
+            };
+
+            trace!(
+                "{} - request_id to look up to correlate to req_id: {}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_VSOMEIP_MSG_TO_UMSG,
+                request_id
+            );
+            let mut ue_request_correlation = UE_REQUEST_CORRELATION.write().await;
+            let Some(req_id) = ue_request_correlation.remove(&request_id) else {
+                return Err(UStatus::fail_with_code(
+                    UCode::NOT_FOUND,
+                    format!(
+                        "Corresponding reqid not found for this SOME/IP RESPONSE: {}",
+                        request_id
+                    ),
+                ));
+            };
+
+            let umsg_res = UMessageBuilder::response(sink, req_id, source)
+                .with_comm_status(UCode::OK.value())
+                .build_with_payload(payload_bytes, UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED);
+
+            let Ok(umsg) = umsg_res else {
+                return Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    "Unable to build UMessage from vsomeip message",
+                ));
+            };
+
+            Ok(umsg)
+        }
+        message_type_e::MT_ERROR => {
+            trace!("MT_ERROR type");
+            let sink = UUri {
+                authority_name,
+                ue_id: client_id as u32,
+                ue_version_major: 1, // TODO: I don't see a way to get this
```

Comment:
I don't see a way to obtain the ue_version_major for the sink for a SOME/IP / vsomeip MT_ERROR, so I punted and set to 1 for now.

Any thoughts @stevenhartley / @int0x27?